core/trino-main/src/main/java/io/trino/execution/SqlTaskExecution.java

```diff
@@ -589,7 +589,6 @@ public void enqueueSplits(Set<ScheduledSplit> splits, boolean noMoreSplits)
             verify(driverFactory.getSourceId().isPresent(), "not a source driver");
             verify(!this.noMoreSplits.get() || splits.isEmpty(), "cannot add splits after noMoreSplits is set");
             queuedSplits.addAll(splits);
-            verify(!this.noMoreSplits.get() || noMoreSplits, "cannot unset noMoreSplits");
```

Comment:
Would you like to add a document here to explain the problem?

Reply:
Yeah - please add a sentence of explanation that here we can get non empty splits list even if we already transitioned to `noMoreSplits` state, and this is not a big deal as those will be filtered out later.
Actually we can probably drop it here already.

Reply:
Oh wait - those are filtered out before, not after the call :) Please just note that in the comment.